### ISSUE TITLE
#Textsum model - System Freeze running Textsum training after 1.0.1 conversion 

### ISSUE THREAD

**Issue by xtr33me** [tensorflow/models#1215]
I'm unsure if this is due to something new that has changed in TF1.0+ however when I run the textsum model with the "Train" mode set, my system will freeze at about the location shown in the screenshot below. It seems to occur once it has stopped raising the pool size limit.  When I say freeze, I mean that my system becomes unusable. The only thing I am able to do is move my mouse and it will move ever so slowly, but I can't do much of anything else.  It almost seems that it uses all my system resources in 1.0.1 whereas when I run it using the same settings in 0.11, I have no issues at all.  Once my system freezes, no more training occurs and I have to hard reboot. 

If at all possible, can someone please try to train this model against the original LDC dataset or a larger one than the 20 articles and see if the same thing is happening? 

Now if I run this in my TF 0.11 setup, everything works fine and training will run non-stop for weeks.  The dataset I am running on is about 1.3mil articles. When I run the textsum model against the included "play" dataset of 20 articles, I don't seem to run into this issue.

I am running this in Ubuntu 16.04, Cuda 8.0, CuDnn 5.1.10, TF 1.0.1 and my GPU is: GeForce GTX 980M.  Also during the Tensorflow ./configure I tried both disabling and enabling "jemalloc as the malloc implementation".  Currently I have it enabled whereas this was not an option when I was using 0.11 (I dont believe...and if it was I did not enable it).

nvidia-smi output after breaking out of training epoch, but before lockup.
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 375.39                 Driver Version: 375.39                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 980M    Off  | 0000:01:00.0     Off |                  N/A |
| N/A   46C    P8     8W /  N/A |    324MiB /  8121MiB |      2%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0      1159    G   /usr/lib/xorg/Xorg                             225MiB |
|    0      1637    G   compiz                                          64MiB |
|    0      2256    G   ...s-passed-by-fd --v8-snapshot-passed-by-fd     6MiB |
|    0      3042    G   ...s-passed-by-fd --v8-snapshot-passed-by-fd    26MiB |
+-----------------------------------------------------------------------------+

![screenshot from 2017-03-19 09-23-59](https://cloud.githubusercontent.com/assets/1653419/24082221/3c47952c-0c98-11e7-9067-250b3b14b3d3.png)

Right before my system locked up, more of the CPU's went further up (about 40-50% on avg) and memory was sitting at 15.2G/15.5G.  The snapshot below was taken somewhat before my whole system locked. Unsure if it helps, but figured I'd share.

![screenshot from 2017-03-22 21-24-02](https://cloud.githubusercontent.com/assets/1653419/24227859/b947e47c-0f46-11e7-9ffb-c92b606c134b.png)

**Comment by skye** [comment]
@panyx0718 or @peterjliu, can one of you take a look at this?

**Comment by xtr33me** [comment]
I got some time to upgrade to the latest and after changing over all references of CPU0 to GPU0 again, I do not seem to be having this issue any longer.  Going to close the ticket.

**Comment by AngJianHui** [comment]
@xtr33me  Hi, I am facing the same issue. Could you share with me how you solve it?

**Comment by xtr33me** [comment]
@AngJianHui In seq2seq_attention in the _Train function I changed the entry for device to /gpu:0 rather than /cpu:0.  I also changed all references for the same thing in the seq2seq_attention_model.py file as well.  I still had some crazy allocation on virtual memory as shown in the screenshot but I have been at least able to train for the past week non-stop and haven't had any issues.  That said, there is still something going on with the allocation of resources I think because if I attempt to run Tensorboard while training is going on, everything freezes again.  I haven't found a solution for that as of yet, but if I just spawn up the training from commandline, all is working fine for me now. It's just hands-off for anything else while I'm doing so.  Hope this helps you some.
